### ISSUE TITLE
Including Implicit negative feedback

### ISSUE THREAD

**Issue by patrickorlando** [tensorflow/recommenders#272]
Hi,
Thanks for this great library!

This is a general question for some guidance or literature on the topic of implicit negative feedback.
The dataset I have relates to audio tracks. There can be various signals of implicit interest but the simplest and strongest is how much they listened to the track. Multiple times indicates strong preference whilst skipping is a pretty clear signal of dislike.

It would seem reasonable to assume that including this behaviour would improve the recommendations, but I can't seem to work out how to include it in a retrieval type model.

I've attempted including these in a multi-task learner, where the sample weight in the retrieval model is set low for these dislike examples, however it results some oscillations in the training loss, and ultimately worse recall from the retrieval model.

Are there any examples or ideas on how to do this?
Thanks in advanced!

**Comment by maciejkula** [comment]
What you tried seems like a good idea. You could even give disliked examples a negative weight, to indicate to the model that they should be avoided?

In most cases you have a two stage system, with a retrieval first stage and a ranking second stage. Incorporating this kind of feedback is somewhat easier in the second stage.

**Comment by jashshah** [comment]
@maciejkula can you share a code snippet of how we can use  `sample weight`? Would it be used in the `compute_loss` method?

Something like this:

 ```
def compute_loss(self, features: Dict[Text, tf.Tensor], training=False) -> tf.Tensor:
    # We pick out the user features and pass them into the user model.
    user_embeddings = self.user_model(features["user_id"])
    # And pick out the movie features and pass them into the movie model,
    # getting embeddings back.
    positive_movie_embeddings = self.movie_model(features["movie_title"])

    # The task computes the loss and the metrics.
    return self.task(user_embeddings, positive_movie_embeddings, sample_weights)
```

Would we have to return a third object from the `tf.data` pipeline along with users and movies?
